### ISSUE TITLE
Candidate values are permanently `null` when getting datastore entries

### ISSUE THREAD

**Issue by BatiGencho** [massalabs/massa#2784]
Candidate values when getting datastore entries are always  null whereas the final ones are correctly  updated. The new pub json rpc method (get_datastore_entries) correctly delivers final values, but none-final (candidate ones) are null all the time. The relevant js method for getting those is as of this moment:

```ts    public async getDatastoreEntries(addresses_keys: Array<IDatastoreEntryInput>): Promise<Array<IContractStorageData>> {
        let data = [];
        for (let input of addresses_keys) {
            data.push({
                address: input.address,
                key: Array.prototype.slice.call(Buffer.from(input.key))
            });
        }
        const jsonRpcRequestMethod = JSON_RPC_REQUEST_METHOD.GET_DATASTORE_ENTRIES;
        if (this.clientConfig.retryStrategyOn) {
            var datastoreEntries = await trySafeExecute<Array<IDatastoreEntry>>(this.sendJsonRPCRequest, [jsonRpcRequestMethod, [data]]);
        } else {
            var datastoreEntries = await this.sendJsonRPCRequest<Array<IDatastoreEntry>>(jsonRpcRequestMethod, [data]);
        }
        const candidateDatastoreEntries: Array<Array<number>|null> = datastoreEntries.map(elem => elem.active_value);
        const finalDatastoreEntries: Array<Array<number>|null> = datastoreEntries.map(elem => elem.final_value);
        return datastoreEntries.map((_, index) => {
            return {
                candidate: ( candidateDatastoreEntries[index] ) ? candidateDatastoreEntries[index].map((s) => String.fromCharCode(s)).join(""): null,
                final: ( finalDatastoreEntries[index] ) ? finalDatastoreEntries[index].map((s) => String.fromCharCode(s)).join(""): null
            } as IContractStorageData;
        });```

**Comment by damip** [comment]
@Eitu33  can you take a look ?

**Comment by Eitu33** [comment]
On it

**Comment by Eitu33** [comment]
I've been testing the endpoint more in-depth to locate the issue but I'm not able to reproduce, candidate values are correctly updated.

I've been using this SC: https://github.com/massalabs/massa-sc-examples/tree/improve-async-example/async-calls/src
And the `get_datastore_entry` CLI cmd that I've implemented for the occasion: https://github.com/massalabs/massa/pull/2786

I will keep testing to see if I can find something.

**Comment by damip** [comment]
> I've been testing the endpoint more in-depth to locate the issue but I'm not able to reproduce, candidate values are correctly updated.
> 
> I've been using this SC: https://github.com/massalabs/massa-sc-examples/tree/improve-async-example/async-calls/src And the `get_datastore_entry` CLI cmd that I've implemented for the occasion: #2786
> 
> I will keep testing to see if I can find something.

For info, other partners have also pinged us with similar complaints

**Comment by damip** [comment]
> I've been testing the endpoint more in-depth to locate the issue but I'm not able to reproduce, candidate values are correctly updated.
> 
> I've been using this SC: https://github.com/massalabs/massa-sc-examples/tree/improve-async-example/async-calls/src And the `get_datastore_entry` CLI cmd that I've implemented for the occasion: #2786
> 
> I will keep testing to see if I can find something.

@Eitu33  For info, other partners have also pinged us with similar complaints

**Comment by Eitu33** [comment]
Issue came from the naming update I've made on the endpoint payload, from `active_value` to `candidate_value`. Evgeni was not aware of the change and the `massa-web3` was not up to date. It's my mistake for not letting him know.
